### PR TITLE
Feature/addsMarketsandStoresFilterButtons

### PR DESCRIPTION
**Pull request by cordhernandez**
+ adjusts the buttons border width
+ adjusts the buttons corner radius
+ adjusts the buttons background color
+ calls styleMenu() in viewDidLoad()
+ the logic/delegate code for these buttons will be added in another ticket
#143

### CODE REVIEW

File: BlackNectar/BlackNectar/FilterViewController.swift [hunk 1]
```diff
@@ -216,16 +263,30 @@ extension FilterViewController {
...
+//MARK: Style Menu Code
 extension FilterViewController {
     
-    func mapView(_ mapView: MKMapView, regionDidChangeAnimated animated: Bool) {
+    func styleMenu() {
         
-        let center = mapView.centerCoordinate
+        farmersMarketsButton.layer.borderColor = UIColor.white.cgColor
+        farmersMarketsButton.layer.borderWidth = 1
+        farmersMarketsButton.layer.cornerRadius = 7
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/FilterViewController.swift]: This is the sort of thing that can be done from Storyboard by using `@IBDesignable`.
It is recommended to do visual stuff in the visual editor, to maximize productivity and minimize code footprint.

Also, the cornerRadius is supposed to be 5. :wink:

File: BlackNectar/BlackNectar/FilterViewController.swift [hunk 2]
```diff
@@ -216,16 +263,30 @@ extension FilterViewController {
...
+        groceryStoresButton.layer.borderColor = UIColor.white.cgColor
+        groceryStoresButton.layer.borderWidth = 1
+        groceryStoresButton.layer.cornerRadius = 7
         
-        self.loadStoresInMapView(at: center)
+    }
+    
+    func styleButtonOn(button: UIButton) {
+        
+        button.layer.backgroundColor = UIColor.init(red: 0.902, green: 0.73, blue: 0.25, alpha: 1).cgColor
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/FilterViewController.swift]: These color values seem guessed at. You definitely want to use code that lets you put in colors using HEX value.

Check out https://github.com/RedRoma/RedRomaColors:

https://github.com/RedRoma/RedRomaColors/blob/develop/RedRomaColors/RedRomaColors/Colors.swift

`Colors.fromRGB()`
`Colors.from(hexString:"")`

**Comment by cordhernandez** [comment]
need an admin to check this branch. thanks.

File: BlackNectar/BlackNectar/Colors.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+//  Created by Cordero Hernandez on 2/8/17.
+//  Copyright © 2017 BlackSource. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public class Colors {
+    
+    public static func fromRGB(red: CGFloat, green: CGFloat, blue: CGFloat) -> UIColor {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/Colors.swift:14]: Nice!

File: BlackNectar/BlackNectar/FilterViewController.swift [hunk 4]
```diff
@@ -17,12 +17,14 @@ import UIKit
...
     @IBOutlet weak var mapView: MKMapView!
+    @IBOutlet weak var farmersMarketsButton: UIButton!
+    @IBOutlet weak var groceryStoresButton: UIButton!
     
     var currentCoordinates: CLLocationCoordinate2D?
     
     var distance = 0.0
-    var showFarmersMarkets = true
-    var showGroceryStores = true
+    var showFarmersMarkets = false
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/FilterViewController.swift:26]: The best option would be to load whatever the user has to set to default from `UserDefaults`

File: BlackNectar/BlackNectar/FilterViewController.swift [hunk 5]
```diff
@@ -216,16 +262,21 @@ extension FilterViewController {
...
     
-    func mapView(_ mapView: MKMapView, regionDidChangeAnimated animated: Bool) {
-        
-        let center = mapView.centerCoordinate
+    func styleButtonOn(button: UIButton) {
         
-        LOG.debug("User dragged Map Screen to: \(center)")
+        button.backgroundColor = Colors.fromRGB(red: 235, green: 191, blue: 77)
+        button.titleLabel?.font = UIFont(name: "Oxygen-Bold", size: 16)
+        UIView.transition(with: button, duration: 0.4, options: .transitionCrossDissolve, animations: nil, completion: nil)
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/FilterViewController.swift]: Technically....the code that changes the buttons should be in an animations closure. That closure would then be passed to the UIView.transition method.

```swift
let animations = {
    button.backgroundColor = Colors.fromRGB(red: 235, green: 191, blue: 77)
    button.titleLabel?.font = UIFont(name: "Oxygen-Bold", size: 16)
}
UIView.transition(with: button, duration: 0.4, options: .transitionCrossDissolve, animations: animations, completion: nil)
```

File: BlackNectar/BlackNectar/FilterViewController.swift [hunk 6]
```diff
@@ -216,16 +262,21 @@ extension FilterViewController {
...
+    }
+    
+    func styleButtonOff(button: UIButton) {
         
-        self.loadStoresInMapView(at: center)
+        button.backgroundColor = UIColor.clear
+        button.titleLabel?.font = UIFont(name: "Oxygen-Regular", size: 16)
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/FilterViewController.swift]: These fonts should ideally be in their own class `Fonts`, so  you can quickly load them as variables, and not have to enter the String name here all the time.

**Review by SirWellington — APPROVED** [review]
:green_apple:

File: BlackNectar/BlackNectar/Colors.swift (hunk 3 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/Colors.swift:14]: 👊🏾

File: BlackNectar/BlackNectar/FilterViewController.swift (hunk 4 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/FilterViewController.swift:26]: will do this in ticket #149

File: BlackNectar/BlackNectar/FilterViewController.swift (hunk 6 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/FilterViewController.swift]: alright now it looks clean 
job done

File: BlackNectar/BlackNectar/FilterViewController.swift (hunk 5 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/FilterViewController.swift]: Ohh ok this makes sense 
better to put animations instead of nil lol

Comment (cordhernandez) [reply BlackNectar/BlackNectar/FilterViewController.swift]: job done

**Comment by cordhernandez** [comment]
changes have been made 
please check 
thanks

File: BlackNectar/BlackNectar/Fonts.swift [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+import Foundation
+import UIKit
+
+class Fonts {
+    
+    static let instance = Fonts()
+    private init() {}
+    
+    var oxygenBold = UIFont(name: "Oxygen-Bold", size: 16)
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/Fonts.swift]: Wonderful!
You could probably skip the whole singleton pattern here and just have these be static constant variables.
`Fonts.oxygenBold`, etc

File: BlackNectar/BlackNectar/FilterViewController.swift [hunk 8]
```diff
@@ -216,16 +262,29 @@ extension FilterViewController {
...
+        UIView.transition(with: button, duration: 0.4, options: .transitionCrossDissolve, animations: animations, completion: nil)
         
-        self.loadStoresInMapView(at: center)
+    }
+    
+    func styleButtonOff(button: UIButton) {
+        
+        let animations = {
+            button.backgroundColor = UIColor.clear
+            button.titleLabel?.font = Fonts.instance.oxygenRegular
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/FilterViewController.swift]: This is much betta.

**Review by SirWellington — COMMENTED** [review]
One minor tweak and we good

File: BlackNectar/BlackNectar/Fonts.swift (hunk 7 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/Fonts.swift]: ahh that's what it was 
cool this is done

File: BlackNectar/BlackNectar/FilterViewController.swift (hunk 8 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/FilterViewController.swift]: 👍🏽

**Pull request merged**